base/iostream.jl

```diff
@@ -321,13 +321,18 @@ function peek(s::IOStream)
     ccall(:ios_peekc, Cint, (Ptr{Void},), s)
 end
 
+
+function _seekback(s::IOStream, pos::Integer)
+    ccall(:ios_seekback, Cint, (Ptr{Void}, Cint), s, pos)
+end
+
 function skipchars(io::IOStream, pred; linecomment=nothing)
     while !eof(io)
         c = read(io, Char)
         if c === linecomment
             readline(io)
         elseif !pred(c)
-            seek(io,position(io)-sizeof(string(c)))
+            _seekback(io, codelen(c))
```

Comment:
Well, now don't I feel silly, I was sure I looked through all the IO functions before resorting to this. That's way better, I was feeling guilty for having introduced a C function when I barely use the language (and I probably did it wrong too, didn't I?)

I will use that then.